src/builder/command.rs

```diff
@@ -708,6 +708,52 @@ impl<'help> App<'help> {
         c.print()
     }
 
+    /// Returns the help message (`-h`).
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use clap::Command;
+    /// let mut cmd = Command::new("myprog");
+    /// let help: String = cmd.render_help();
+    /// print!("{}",help);
+    /// ```
+    pub fn render_help(&mut self) -> String {
+        self._build_self();
+
+        let mut help: String = "".into();
+        let usage = Usage::new(self);
+        match Help::new(HelpWriter::String(&mut help), self, &usage, false).write_help() {
+            Ok(_) => help,
+            Err(e) => {
+                panic!("{}", e.to_string())
+            }
+        }
+    }
+
+    /// Returns the long help message.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use clap::Command;
+    /// let mut cmd = Command::new("myprog");
+    /// let help: String = cmd.render_long_help();
+    /// print!("{}",help);
+    /// ```
+    pub fn render_long_help(&mut self) -> String {
+        self._build_self();
+
+        let mut help: String = "".into();
+        let usage = Usage::new(self);
+        match Help::new(HelpWriter::String(&mut help), self, &usage, true).write_help() {
+            Ok(_) => help,
+            Err(e) => {
+                panic!("{}", e.to_string())
+            }
```

Comment:
We should either be returning an error or making it clear why a panic is unexpected to happen.  The latter is the case for this.  We have a common panic message for these cases (its `INTERNAL` something I think).  We can just `write_help().expect(INTERNAL_...)`